Assets/Scripts/AI/AIManager.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class AiManager : MonoBehaviour
+{
+    private NavMeshAgent agent;
+    private Animator animator;
+    public Transform player;
+
+    public void Awake()
+    {
+        agent = GetComponent<NavMeshAgent>();
+        animator = GetComponent<Animator>();
```

Comment:
If certain components are required for the AI manager to work, add a `RequiredComponent` property at the top of the class. This will ensure that the components you're trying to get on `Awake` (or anywhere else) exists.

Reply:
Sure thing